src/Umbraco.Web/Strategies/RelateOnTrashHandler.cs

```diff
@@ -0,0 +1,65 @@
+﻿using System;
+using System.Linq;
+using Umbraco.Core;
+using Umbraco.Core.Auditing;
+using Umbraco.Core.Events;
+using Umbraco.Core.Models;
+using Umbraco.Core.Services;
+
+namespace Umbraco.Web.Strategies
+{
+    public sealed class RelateOnTrashHandler : ApplicationEventHandler
+    {
+        protected override void ApplicationStarting(UmbracoApplicationBase umbracoApplication, ApplicationContext applicationContext)
```

Comment:
One thing to consider with that is whether the creation of those relations should raise Saving/Saved events of its own, which isn't possible if its done at the repository level.
Maybe its implicit that a specific relation is created during another operation, but it also means that you can't directly use it in something like Courier. Unless maybe you add additional info to the event args.

Just something to consider - kinda related to the Macro thing I mentioned on Slack.